tests/profiles-enabled/test_shipment_tags.py

```diff
@@ -0,0 +1,223 @@
+#  Copyright 2020 ShipChain, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import uuid
+
+import pytest
+from rest_framework.reverse import reverse
+
+from apps.shipments.models import ShipmentTag
+
+
+@pytest.fixture
+def shipment_tag_creation_data():
+    return {
+        'tag_type': '*Export-Shipment*',
+        'tag_value': '-->Europe'
+    }
+
+@pytest.fixture
+def shipment_tag_location():
+    return {
+        'tag_type': 'Destination',
+        'tag_value': 'Toronto'
+    }
+
+@pytest.fixture
+def missing_tag_type_creation_data():
+    return {
+        'tag_value': '-->Europe'
+    }
+
+@pytest.fixture
+def missing_tag_value_creation_data():
+    return {
+        'tag_type': '*Export-Shipment*',
+    }
+
+@pytest.fixture
+def space_in_tag_type_creation_data():
+    return {
+        'tag_type': '*Export Shipment*',
+        'tag_value': '-->Europe'
+    }
+
+@pytest.fixture
+def space_in_tag_value_creation_data():
+    return {
+        'tag_type': '*Export-Shipment*',
+        'tag_value': 'United Kingdom'
+    }
+
+@pytest.fixture
+def shipment_tags(org_id, shipment, second_shipment, shipment_tag_creation_data, shipment_tag_location):
+    tags = []
+    tags.append(ShipmentTag.objects.create(tag_type=shipment_tag_creation_data['tag_type'],
+                                           tag_value=shipment_tag_creation_data['tag_value'],
+                                           shipment_id=shipment.id,
+                                           owner_id=uuid.UUID(org_id)))
+
+    tags.append(ShipmentTag.objects.create(tag_type=shipment_tag_location['tag_type'],
+                                           tag_value=shipment_tag_location['tag_value'],
+                                           shipment_id=second_shipment.id,
+                                           owner_id=uuid.UUID(org_id)))
+
+    return tags
+
+@pytest.fixture
+def entity_shipment_tagged(json_asserter, shipment, shipment_tags):
```

Comment:
I'd recommend not using fixtures in this way. It has a negative impact on the readability of our unit tests - you can't tell what assertions are being made just by skimming the unit test. I think the json_asserter tool itself does a great job of DRYing assertions, no need to obfuscate our tests further.

Reply:
One more note, I think the preferred method for using the json assert logic is now directly via the class instead of the fixture.  Initially it was developed as a fixture, but by the time it was completed, we didn't really _need_ it to be a fixture.  See an example here: https://github.com/ShipChain/transmission/blob/1a51d60e7d6685f6223b94adecb762cddda92b74/tests/profiles-enabled/test_shipments_states.py#L219